### PR TITLE
GH-426: Modify projects pages (resolves #426)

### LINKED ISSUE #426: Allow projects to have individual pages with possible sub-pages
## Is your feature request related to a problem? Please describe.

Currently if a project requires more detailed information on the IDRC site, a separate page is created in `src/pages/` and then linked from the Projects and Tools page. This is not an ideal solution as projects should not be in the pages directory, and projects may require more pages.

## Describe the solution you'd like

Projects requiring dedicated pages should reside in the `src/projects/` directory. If a project requires additional (sub)pages (like for project reports), then there should be a way to include those (by possibly using sub-directories).

## Additional context

Currently the EDS project has a page under `src/pages/future-of-work-equitable-digital-systems.md`. Eventually this project will need to host project reports in HTML format and other content.

Previously for the IUI project, guidelines were created for that project (HTML, DOCX, and PDF) but they all had to be hosted on the wiki and linked from the IUI project page. It would have been better if all that content were to be hosted on the IDRC website and used sub pages for the HTML version of the guideline.

### PR DESCRIPTION
**Pull request by cindyli**
* [X] This pull request has been tested by running `npm run test` without errors
* [X] This pull request has been built by running `npm run build` without errors
* [X] This isn't a duplicate of an existing pull request

## Description

This pull request implements:
1. The new landing page for the "Projects" page;
2. Allow each project to have sub-pages.

## Steps to test

1. Open the deploy preview and go to "Projects";
2. The landing page should match the new design;
3. Open the admin page to author multiple pages for a project. Subpages should select "parent page title";
4. Go back to the "Projects" page and select this project. The hierarchy of pages should present in the way that's described in the https://github.com/inclusive-design/idrc/issues/426.

**Expected behavior:** 
Described above.

### CODE REVIEW

**Comment by jhung** [comment]
I was testing and have some questions and found some issues:

Issues:
- Project listing description is mandatory, but should be optional if a parent page is specified. Otherwise user will get an error when trying to submit a new sub page and may input garbage data to avoid the error which may lead to confusion down the road.
- Project body is mandatory, but should be optional if there is a project listing description for situations for projects with no internal page (and just a link to an external website).
- Sub page ordering does not appear to work despite specifying a value.

Improvements:
- Add help text to the Title field: "Suggestion: use a descriptive title which includes keywords from the project."
- Change "Short Name" to "Project Short Name"
- Add help text to Short Name field: "Short name for the project. This is used to generate the name for external project website links."
- Add help text to the Project Name field: "Suggestion: If this is a subpage of a project, enter the name of the project here."

**Comment by cindyli** [comment]
Thanks for great suggestions, @jhung. The last commit should address them all.

Regarding "Sub page ordering does not appear to work despite specifying a value", I cannot reproduce the problem. The child pages are listed on the side navigation menu in the ascending order of "subpage order" value - the lowest order number will show at the top while the highest at the bottom.

Can you describe steps to reproduce the problem? We can have a call if it makes easier.

**Comment by cindyli** [comment]
@jhung, as we discussed, I removed the `tags` field from the project config and the template. This pull request is ready for a review. Thanks.

File: src/admin/config.yml [hunk 1]
```diff
@@ -429,12 +429,16 @@ collections:
...
-      - {label: "Tags", name: "tags", widget: "list", required: false}
-      - {label: "Link", name: "link", widget: "string", required: true}
+      - {label: "Layout", name: "layout", widget: "hidden", default: "layouts/project.njk"}
+      - {label: "Project Name", name: "projectName", widget: "string", required: true, hint: "Suggestion: If this is a subpage of a project, enter the name of the project here."}
+      - {label: "Title", name: "title", widget: "string", required: true, hint: "Suggestion: use a descriptive title which includes keywords from the project."}
+      - {label: "Project Short Name", name: "shortName", widget: "string", required: false, hint: "Short name for the project. This is used to generate the name for external project website links."}
+      - {label: "Parent Page", name: "parentPageTitle", widget: "relation", collection: "projects", value_field: "title", search_fields: [title], required: false, hint: "Only select for sub-page of a project."}
+      - {label: "Subpage Order", name: "subPageOrder", widget: "number", required: false, hint: "Only select for sub-page of a project."}
```
Comment (jhung) [inline src/admin/config.yml:437]:
```suggestion
      - {label: "Subpage Order", name: "subPageOrder", widget: "number", required: false, hint: "Hint: if using this, all sub-pages at the same level should have an order number specified. Leave blank to have sub-pages appear in alphabetical order. "}
```

File: src/admin/config.yml [hunk 2]
```diff
@@ -429,12 +429,16 @@ collections:
...
+      - {label: "Thumbnail Image", name: "thumbnailImage", widget: "image", required: false}
```
Comment (jhung) [inline src/admin/config.yml:438]:
```suggestion
      - {label: "Thumbnail Image", name: "thumbnailImage", widget: "image", required: false, hint: "Suggested size: 680x494px. Images not fitting this aspect ratio will appear stretched."}
```

File: src/admin/config.yml [hunk 3]
```diff
@@ -429,12 +429,16 @@ collections:
...
+      - {label: "Project Listing Description", name: "description", widget: "markdown", required: false, editor_components: ["image"], hint: "Only write this description for the main page of a project."}
```
Comment (jhung) [inline src/admin/config.yml:439]:
```suggestion
      - {label: "Project Listing Description", name: "description", widget: "markdown", required: false, editor_components: ["image"], hint: "This is text that appears for a project in the projects listing. This description is not required for sub-pages."}
```

File: src/admin/config.yml [hunk 4]
```diff
@@ -429,12 +429,16 @@ collections:
...
+      - {label: "External Project Website", name: "link", widget: "string", required: false}
```
Comment (jhung) [inline src/admin/config.yml:440]:
```suggestion
      - {label: "External Project Website", name: "link", widget: "string", required: false, hint: "This is not required for sub-pages."}
```

File: src/admin/config.yml [hunk 5]
```diff
@@ -429,12 +429,16 @@ collections:
...
+      - {label: "Body", name: "body", required: false, widget: "markdown"}
```
Comment (jhung) [inline src/admin/config.yml:441]:
```suggestion
      - {label: "Body", name: "body", required: false, widget: "markdown", hint: "This is the content that appears for a project page hosted on the IDRC site."}
```

File: src/admin/config.yml [hunk 6]
```diff
@@ -429,12 +429,16 @@ collections:
...
     create: true
     fields:
-      - {label: "Layout", name: "layout", widget: "hidden", default: "layouts/single--project.njk"}
-      - {label: "Title", name: "title", widget: "string", required: true}
-      - {label: "Short Name", name: "shortName", widget: "string", required: false}
-      - {label: "Description", name: "description", widget: "markdown", required: true, editor_components: ["image"]}
-      - {label: "Tags", name: "tags", widget: "list", required: false}
-      - {label: "Link", name: "link", widget: "string", required: true}
+      - {label: "Layout", name: "layout", widget: "hidden", default: "layouts/project.njk"}
+      - {label: "Project Name", name: "projectName", widget: "string", required: true, hint: "Suggestion: If this is a subpage of a project, enter the name of the project here."}
```
Comment (jhung) [inline src/admin/config.yml:433]:
```suggestion
      - {label: "Project Name", name: "projectName", widget: "string", required: true, hint: "Suggestion: If this is a sub-page of a project, enter the name of the project here."}
```

File: src/admin/config.yml [hunk 7]
```diff
@@ -429,12 +429,16 @@ collections:
...
+      - {label: "Title", name: "title", widget: "string", required: true, hint: "Suggestion: use a descriptive title which includes keywords from the project."}
```
Comment (jhung) [inline src/admin/config.yml:434]:
```suggestion
      - {label: "Body Title", name: "title", widget: "string", required: true, hint: "The title that appears after the Project Name and before the Body content. Suggestion: use a descriptive title which includes keywords from the project."}
```

**Review by jhung — CHANGES_REQUESTED** [review]
I made some suggested changes for the content editor labels and hint text.

No other issues found.

**Review by jhung — APPROVED** [review]

**Pull request merged**